app/src/main/java/org/greenstand/android/TreeTracker/database/entity/UserEntity.kt

```diff
@@ -36,6 +36,9 @@ data class UserEntity(
     @ColumnInfo(name =
     "power_user", defaultValue = "0")
     var powerUser: Boolean,
+    @ColumnInfo(name =
+    "unread_messages_available", defaultValue = "0")
+    var unreadMessagesAvailable: Boolean,
```

Comment:
I think this makes the code more complicated than it needs to be. We know which messages are read, and which user they belong to. We can easily query how many messages a user has with that info without keeping track in the user object. 
When you place this info in the user object, it's more data that has to now keep in sync.

Reply:
The user object updates this screen 
![Screenshot_20220612_133712](https://user-images.githubusercontent.com/38661872/173233753-5fdbe4a2-7121-4448-9de5-38d4c623db63.png)
so it is easier to have it as a value in the User object which updates each user button on the screen.

Reply:
I agree, so we can add it to the user object, but not the user entity. This is why we have a repo layer, we can separate how we store things with what the rest of the app has access to. We can query the messages when returning a user object to fill that fiels.

Reply:
@Elforama sorry I was away for a while.  Yeah this makes more sense actually.